eng/build.ps1

```diff
@@ -20,46 +20,78 @@ Param(
-  Write-Host "  -rebuild                Rebuild all source projects"
-  Write-Host "  -test                   Build and run tests (short: -t)"
-  Write-Host "  -pack                   Package build outputs into NuGet packages"
-  Write-Host "  -sign                   Sign build outputs"
-  Write-Host "  -publish                Publish artifacts (e.g. symbols)"
-  Write-Host "  -clean                  Clean the solution"
+  Write-Host "  -build (-b)             Build all source projects."
+  Write-Host "  -clean                  Clean the solution."
+  Write-Host "  -pack                   Package build outputs into NuGet packages."
+  Write-Host "  -publish                Publish artifacts (e.g. symbols)."
+  Write-Host "  -rebuild                Rebuild all source projects."
+  Write-Host "  -restore                Restore dependencies."
+  Write-Host "  -sign                   Sign build outputs."
+  Write-Host "  -test (-t)              Build and run tests."
   Write-Host ""
 
   Write-Host "Libraries settings:"
-  Write-Host "  -framework              Build framework: net5.0 or net472 (short: -f)"
-  Write-Host "  -coverage               Collect code coverage when testing"
-  Write-Host "  -testscope              Scope tests, allowed values: innerloop, outerloop, all"
-  Write-Host "  -testnobuild            Skip building tests when invoking -test"
-  Write-Host "  -allconfigurations      Build packages for all build configurations"
+  Write-Host "  -allconfigurations      Build packages for all build configurations."
+  Write-Host "  -coverage               Collect code coverage when testing."
+  Write-Host "  -framework (-f)         Build framework: net5.0 or net472."
+  Write-Host "  -testnobuild            Skip building tests when invoking -test."
+  Write-Host "  -testscope              Scope tests, allowed values: innerloop, outerloop, all."
   Write-Host ""
 
-  Write-Host "Command-line arguments not listed above are passed thru to msbuild."
-  Write-Host "The above arguments can be shortened as much as to be unambiguous (e.g. -con for configuration, -t for test, etc.)."
+  Write-Host "Command-line arguments not listed above are passed through to MSBuild."
+  Write-Host "The above arguments can be shortened as much as to be unambiguous."
+  Write-Host "(Example: -con for configuration, -t for test, etc.)."
+  Write-Host ""
+
+  Write-Host "Here are some quick examples:"
+  Write-Host ""
+  Write-Host "Build CoreCLR on Windows for x64 on release configuration:"
+  Write-Host ".\build.cmd clr -os Windows_NT -arch x64 -configuration Release"
+  Write-Host ""
+  Write-Host "Cross-compile CoreCLR runtime on Windows for ARM64."
+  Write-Host ".\build.cmd -subset clr.runtime -os Windows_NT -arch arm64 -configuration Release"
+  Write-Host ""
+  Write-Host "Build Debug libraries with a Release runtime."
+  Write-Host ".\build.cmd -subset clr+libs -os Windows_NT -arch x64 -runtimeConfiguration Release"
```

Comment:
This is super verbose and I don't know anyone specifying the defaults for OS and Arch.

```suggestion
  Write-Host ".\build.cmd clr+libs -rc Release"
```

Reply:
I agree, I never specify -os or -arch explicitly.

Reply:
I mean, I do specify arch sometime when I want to do a cross build, but it is very rare.

Reply:
I often specify arch when building on Windows x64 to build arm/arm64/x86.

One thing that is very confusing to me is: when do I need to specify `-cross`?

Reply:
From what the docs say and my past experiences, `--cross` is a flag specified when you are doing a cross-build. Worth mentioning it's only necessary on Unix (i.e. `build.sh`)